### DISCUSSION TITLE
Event driven expressions

### DISCUSSION THREAD

**Discussion by eitancaspi** [Cookie-AutoDelete/Cookie-AutoDelete#1307] [Ideas / Feature Requests]
Hello,

Thank you for an AWSOME extension! it is really helpful.

I wish to ask to make it even better by adding to the expressions the ability to set, per domain definition, that the object types that were selected to be deleted - will be deleted by events triggers - like page refresh, moving to another address on the same domain, simply time recurrence (e.g. every x seconds/minutes) and so on.

Thank you!!

**Comment by kennethtran93** [comment]
This is essentially what lists are for as they are also event specific as well (though some options are toggled through global settings).

I'll keep in mind the page refresh (not sure how to track that), separate page on same domain (possible, but by the time the new page finishes loading it would probably have fetched existing cookies before we would have a chance to remove them).  The time recurrence one has been mentioned previously.

**Reply by eitancaspi** [reply]
Thank you Ken.

Well, the list of expressions are currently limited for the user to select which types of data to remove and only if either never to touch them (whitelist) or remove them only when the browser closes.
It will be nicer to make it more granular to also automate the deletion when using the browser, with the options I mentioned.

Thank you VERY MUCH! :)

**Reply by kennethtran93** [reply]
It has been already requested to have more 'events'.  Currently you can also clean on domain change when enabled.

**Reply by eitancaspi** [reply]
Yeah, I know of the domain change one, but I wish, and so it seems - others, to have more events while on the current domain. Crossing my fingers with hope for quick realization... :)
